core-management/src/main/java/org/wildfly/extension/core/management/CoreManagementSubsystemParser_1_0.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2016, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.wildfly.extension.core.management;
+
+import static org.jboss.as.controller.PersistentResourceXMLDescription.builder;
+
+import java.util.List;
+
+import javax.xml.stream.XMLStreamConstants;
+import javax.xml.stream.XMLStreamException;
+
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.PersistentResourceXMLDescription;
+import org.jboss.as.controller.persistence.SubsystemMarshallingContext;
+import org.jboss.dmr.ModelNode;
+import org.jboss.staxmapper.XMLElementReader;
+import org.jboss.staxmapper.XMLElementWriter;
+import org.jboss.staxmapper.XMLExtendedStreamReader;
+import org.jboss.staxmapper.XMLExtendedStreamWriter;
+
+/**
+ * Parser and Marshaller for core-management's {@link #NAMESPACE}.
+ *
+ * <em>All resources and attributes must be listed explicitly and not through any collections.</em>
+ * This ensures that if the resource definitions change in later version (e.g. a new attribute is added),
+ * this will have no impact on parsing this specific version of the subsystem.
+ *
+ * @author <a href="http://jmesnil.net/">Jeff Mesnil</a> (c) 2016 Red Hat inc.
+ */
+public class CoreManagementSubsystemParser_1_0 implements XMLStreamConstants, XMLElementReader<List<ModelNode>>, XMLElementWriter<SubsystemMarshallingContext> {
+
+    static final String NAMESPACE = "urn:jboss:domain:core-management:1.0";
+
+    protected static final CoreManagementSubsystemParser_1_0 INSTANCE = new CoreManagementSubsystemParser_1_0();
```

Comment:
Brian knows the details as well but recent experience has showed us that where we build up a resource model and parser entirely from static instances we loose a very powerful aspect of the management model where we can also maintain runtime state independently of services.

For other subsystems this maybe is more of a hypothetical discussion and they can rely completely on services but anything related to management is very :chicken: and :egg:y - as an example for some security related services we have needed to start up relatively unsecured and as the management model is built up we provide new implementation of specific SPIs.

With static references to resources there are still opportunities to maintain some state but the recent case where this broke down was a server reload as the instances were not recreated so state was maintained between server reloads.